### PR TITLE
[Merged by Bors] - feat: `range (f ∘ e) = range f`

### PR DESCRIPTION
**Pull request by YaelDillies**
Match https://github.com/leanprover-community/mathlib/pull/18190

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by fpvandoorn** [comment]
Please fix `Type*` and add an `#align`. Please merge roughly around the same time as the mathlib3 PR merges.

bors d+

**Comment by joneugster** [comment]
I included these changes to `Data.Set.Image` in #2000 which forward-ports a newer change to this file. I'd suggest to close this PR and merge #2000

**Comment by YaelDillies** [comment]
#2000 has to wait on https://github.com/leanprover-community/mathlib/pull/18491, so let's merge this first.

bors merge

File: Mathlib/Data/Set/Basic.lean [hunk 1]
```diff
@@ -4,7 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Jeremy Avigad, Leonardo de Moura
 
 ! This file was ported from Lean 3 source module data.set.basic
-! leanprover-community/mathlib commit b86832321b586c6ac23ef8cdef6a7a27e42b13bd
+! leanprover-community/mathlib commit b875cbb7f2aa2b4c685aaa2f99705689c95322ad
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Basic.lean:7]: Was this intended to be in the other file? https://leanprover-community.github.io/mathlib-port-status/file/data/set/basic?range=b86832321b586c6ac23ef8cdef6a7a27e42b13bd..b875cbb7f2aa2b4c685aaa2f99705689c95322ad lists no changes here.

Comment (eric-wieser) [reply Mathlib/Data/Set/Basic.lean:7]: https://github.com/leanprover-community/mathlib4/pull/2517

**Pull request closed without merging**